File: echo.py
#! /usr/bin/env python3

import argparse

parser = argparse.ArgumentParser()
parser.add_argument("words", nargs="*")
parser.add_argument("-n", action="store_true")

args = parser.parse_args()
print(" ".join(args.words).upper(), end='')
if not args.n:
    print()

File: echo.rs
#!/usr/bin/env -S cargo +nightly -Zscript
---
[package]
edition = "2021"

[dependencies]
clap = { version = "4.2", features = ["derive"] }
---

use clap::Parser;

#[derive(Parser)]
struct Args {
    words: Vec<String>,
    #[arg(short('n'))]
    no_newline: bool,
}

fn main() {
    let args = Args::parse();
    print!("{}", args.words.join(" ").to_uppercase());
    if !args.no_newline {
        println!();
    }
}
